core/src/main/java/com/amplitude/core/utilities/FileResponseHandler.kt

```diff
@@ -65,7 +65,7 @@ class FileResponseHandler(
             removeCallbackByInsertId(eventsString)
             throw e
         }
-        if (eventsList.size == 1) {
+        if (eventsList.size == 1 || badRequestResponse.isInvalidApiKeyResponse()) {
```

Comment:
Here we remove the eventFile from the storage. 
Should we keep the event for an invalid API key instead?

Reply:
You mean keep the event for a valid future API key?

Reply:
Yes, keep it until the next try.  I saw [previous android SDK ](https://github.com/amplitude/Amplitude-Android/blob/main/src/main/java/com/amplitude/api/AmplitudeClient.java#L2337)just logged an error with invalid API.

Reply:
The issue is that we will continuously retry requests that are known to be bad, there's no case in which the instance can recover until restarted with a new API key, which will also effectively orphan these events as storage is also segmented by api key.

Reply:
That makes sense. Thanks.